File: fabric/src/main/java/juuxel/adorn/client/ClientEvents.java
package juuxel.adorn.client;

import juuxel.adorn.client.gui.TradeTooltipComponent;
import juuxel.adorn.client.gui.screen.ModCompatWarningScreen;
import juuxel.adorn.trading.Trade;
import net.fabricmc.fabric.api.client.rendering.v1.TooltipComponentCallback;
import net.fabricmc.fabric.api.client.screen.v1.ScreenEvents;
import net.minecraft.client.gui.screen.TitleScreen;

public final class ClientEvents {
    private static boolean hasSeenTitleScreen = false;

    public static void init() {
        TooltipComponentCallback.EVENT.register(data -> {
            if (data instanceof Trade trade) {
                return new TradeTooltipComponent(trade);
            }

            return null;
        });

        ScreenEvents.BEFORE_INIT.register((client, screen, scaledWidth, scaledHeight) -> {
            if (screen instanceof TitleScreen) {
                if (hasSeenTitleScreen) return;
                hasSeenTitleScreen = true;

                ScreenEvents.beforeTick(screen).register(titleScreen -> {
                    client.setScreen(ModCompatWarningScreen.checkAndCreate(titleScreen, false));
                });
            }
        });
    }
}

File: fabric/src/main/kotlin/juuxel/adorn/client/ClientEvents.kt
package juuxel.adorn.client

import juuxel.adorn.client.gui.TradeTooltipComponent
import juuxel.adorn.trading.Trade
import net.fabricmc.fabric.api.client.rendering.v1.TooltipComponentCallback

object ClientEvents {
    fun init() {
        TooltipComponentCallback.EVENT.register { data ->
            when (data) {
                is Trade -> TradeTooltipComponent(data)
                else -> null
            }
        }
    }
}
